### PR TITLE
fix paging issues on infinite scroll for channels and DMs

### PR DESCRIPTION
**Pull request by patosullivan**
fixes LAND-1250
fixes LAND-1255

### CODE REVIEW

**Comment by latter-bolden** [comment]
Fixes LAND-1255

File: ui/src/chat/ChatWindow.tsx [hunk 1]
```diff
@@ -103,18 +103,26 @@ export default function ChatWindow({
     const { bottom, delayedRead } = useChatStore.getState();
     bottom(true);
     delayedRead(nest, () => markRead({ nest }));
-    if (hasNextPage && !isFetchingNextPage) {
+    if (hasPreviousPage && !isFetchingPreviousPage) {
+      console.log('fetching next page');
```
Comment (arthyn) [inline ui/src/chat/ChatWindow.tsx:107]: should probably reverse language

File: ui/src/chat/ChatWindow.tsx [hunk 2]
```diff
@@ -103,18 +103,26 @@ export default function ChatWindow({
...
+    markRead,
+    fetchPreviousPage,
+    hasPreviousPage,
+    isFetchingPreviousPage,
+  ]);
 
   const onAtTop = useCallback(() => {
-    if (hasPreviousPage && !isFetchingPreviousPage) {
+    if (hasNextPage && !isFetchingNextPage) {
+      console.log('fetching previous page');
```
Comment (arthyn) [inline ui/src/chat/ChatWindow.tsx:121]: should probably reverse language

**Review by arthyn — APPROVED** [review]
lgtm with suggestion

**Pull request merged**